tests/foreman/cli/test_remoteexecution.py

```diff
@@ -941,8 +942,26 @@ def setUpClass(cls):
         super(AnsibleREXTestCase, cls).setUpClass()
         cls.sat6_hostname = settings.server.hostname
         # register and setup a host here and tests will share the host, step 0.
+        cls.org = make_org()
+        cls.client = VirtualMachine(distro=DISTRO_RHEL7)
+        cls.client.create()
+        cls.client.install_katello_ca()
+        cls.client.register_contenthost(
+            org=cls.org['label'],
+            lce='Library'
+        )
+        cls.assertTrue(cls.client.subscribed, True)
+        Settings.set({'name': 'remote_execution_connect_by_ip',
```

Comment:
This doesn't comply with the isolation - you are touching the global settings.
If there was another test relying on this setting, you would directly intefere with it.

- That is why the rest of the tests (with name *_by_ip) rather uses the host parameter. That way you only tell foreman to use ip for the very specific host unique to your test. 

```python
Host.set_parameter({
  'host': self.client.hostname,
  'name': 'remote_execution_connect_by_ip',
  'value': 'True',
})
```

Reply:
thanks @rplevka , I will only change the parameters for that specific host and won't interfere with global settings.